builder/azure/arm/builder.go

```diff
@@ -91,6 +95,18 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	_, _, err = p.ParseUnverified(spnCloud.OAuthToken(), claims)
+
+	if err != nil {
+		return nil, err
+	}
+	b.config.ObjectID = claims["oid"].(string)
```

Comment:
Is this guaranteed?  Do you need to check for an error code?

This (the new block of code) should be it's own method with a good name.

Reply:
Since there is no other user for this, why need a method ? does that aid in readability here?